config/initializers/sentry.rb

```diff
@@ -1,9 +1,24 @@
 # frozen_string_literal: true
 Sentry.init do |config|
   config.environment = HostingEnvironment.environment_name
+  filter_parameters = Rails.application.config.filter_parameters.map(&:to_s)
+  env_var_filters =
+    Regexp.union(
+      ENV
+        .select { |k, _v|
+          filter_parameters.any? { |parameter| k.downcase.include?(parameter) }
+        }
+        .values
```

Comment:
The linters are arguing because we're chaining a select that takes a proc with a `.values` call. You can work around this by using separate variables instead of chaining.

I don't think we need this code however; it wasn't here for my last review.

I understand that it's pre-emptively filtering out all environment variables whose keys don't contain any of the `filter_parameters` we've defined.

This however would miss out on a bunch of secret-y env vars, like:

- `SENTRY_DSN`
- `VCAP_SERVICES`
- `DATABASE_URL`
- `SLACK_WEBHOOK_URL`

Why is it necessary to select from the env vars in this way?

Reply:
On my local host I noticed it was picking up a lot of irrelevant ENV vars causing the Regexp to act erratically when used with the .gsub! method. I suspect this would also produce erratic behaviour on other devs localhost. This was to allow us narrow down our own env vars rather than everything contained in that localhost.

Reply:
I could add _services and _url to the filter params list and it should pick the others. I already added _dsn earlier